### ISSUE TITLE
ccllogic: TestTenantLogicCCL_crdb_internal_tenant is flaky

### ISSUE THREAD

**Issue by yuzefovich** [cockroachdb/cockroach#89230]
I have seen `TestTenantLogicCCL_crdb_internal_tenant` fail several times during CI on unrelated PRs. One example is [here](https://teamcity.cockroachdb.com/viewLog.html?buildId=6762919&buildTypeId=Cockroach_BazelEssentialCi).
```
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8282/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/crdb_internal_tenant:8: SELECT node_id, name FROM crdb_internal.leases ORDER BY name
        expected:
            0  eventlog
            0  jobs
            0  locations
            0  protected_ts_meta
            0  protected_ts_records
            0  public
            0  role_members
            0  role_options
            0  scheduled_jobs
            0  settings
            0  statement_diagnostics_requests
            0  table_statistics
            0  test
            0  users
            
        but found (query options: "") :
            0  eventlog
            0  jobs
            0  join_tokens
            0  locations
            0  migrations
            0  protected_ts_meta
            0  protected_ts_records
            0  public
            0  replication_constraint_stats
            0  replication_stats
            0  role_id_seq
            0  role_members
            0  role_options
            0  scheduled_jobs
            0  settings
            0  span_count
            0  sqlliveness
            0  statement_bundle_chunks
            0  statement_diagnostics_requests
            0  table_statistics
            0  test
            0  ui
            0  users
            0  web_sessions
```
It seems somewhat plausible that #89036 is to blame.

Jira issue: CRDB-20143

**Comment by andrewbaptist** [comment]
Skipping test for now.

**Comment by ajwerner** [comment]
Fixed by #89293
